File: backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ResumeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_profile_id: str
    filename: str
    original_filename: str
    content_type: str | None = None
    file_size_bytes: int
    is_active: bool
    parse_status: str
    parse_error: str | None = None
    parsed_summary: dict | None = None
    skills: list[str] = Field(default_factory=list)
    technologies: list[str] = Field(default_factory=list)
    projects: list[dict] = Field(default_factory=list)
    experience: list[dict] = Field(default_factory=list)
    education: list[dict] = Field(default_factory=list)
    certifications: list[dict] = Field(default_factory=list)
    links: list[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime | None = None
    parsed_at: datetime | None = None


class ResumeListResponse(BaseModel):
    items: list[ResumeResponse]
    total: int
    limit: int
    offset: int


class ResumeUploadResponse(BaseModel):
    resume: ResumeResponse
    warnings: list[str] = Field(default_factory=list)


class ResumeActivateResponse(BaseModel):
    resume_id: str
    is_active: bool
    previous_active_resume_id: str | None = None

File: frontend/types/resume.ts
export type ResumeParseStatus = "uploaded" | "parsed" | "failed" | string;

export type ResumeResponse = {
  id: string;
  user_profile_id?: string | null;
  filename?: string | null;
  original_filename?: string | null;
  content_type?: string | null;
  file_size_bytes?: number | null;
  is_active?: boolean;
  parse_status?: ResumeParseStatus;
  parse_error?: string | null;
  parsed_summary?: Record<string, unknown> | null;
  skills?: string[] | null;
  technologies?: string[] | null;
  projects?: Array<Record<string, unknown>> | null;
  experience?: Array<Record<string, unknown>> | null;
  education?: Array<Record<string, unknown>> | null;
  certifications?: Array<Record<string, unknown>> | null;
  links?: string[] | null;
  created_at?: string | null;
  updated_at?: string | null;
  parsed_at?: string | null;
};

export type ResumeListResponse = {
  items: ResumeResponse[];
  total: number;
  limit: number;
  offset: number;
};

export type ResumeUploadResponse = {
  resume: ResumeResponse;
  warnings?: string[];
};

export type ResumeActivateResponse = {
  resume_id: string;
  is_active: boolean;
  previous_active_resume_id?: string | null;
};

export type ResumeListParams = {
  limit?: number;
  offset?: number;
};
